lib/maplibre.ex

```diff
@@ -151,12 +151,19 @@ defmodule MapLibre do
   """
   @spec add_source(t(), String.t(), keyword()) :: t()
   def add_source(ml, source, opts \\ []) do
-    validate_source_type!(Keyword.get(opts, :type))
+    validate_source!(opts)
     source = %{source => opts_to_ml_props(opts)}
     sources = Map.merge(ml.spec["sources"], source)
     update_in(ml.spec, fn spec -> Map.put(spec, "sources", sources) end)
   end
 
+  defp validate_source!(opts) do
+    type = Keyword.get(opts, :type)
```

Comment:
When you don't need a default value, it's preferable to rely on the Access behavior :)

```suggestion
    type = opts[:type]
```